File: frontend/src/types/script-action.ts
export interface ScriptAction {
  id: number
  name: string
  description?: string | null
  script: string
  param_schema?: Record<string, unknown> | null
  created_at: string
  updated_at: string
}

export interface ScriptActionCreate {
  name: string
  description?: string | null
  script: string
  param_schema?: Record<string, unknown> | null
}

export interface ScriptActionUpdate {
  name?: string
  description?: string | null
  script?: string
  param_schema?: Record<string, unknown> | null
}

File: backend/app/schemas/script_action.py
from datetime import datetime

from pydantic import BaseModel


class ScriptActionCreate(BaseModel):
    name: str
    description: str | None = None
    script: str
    param_schema: dict | None = None


class ScriptActionUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    script: str | None = None
    param_schema: dict | None = None


class ScriptActionRead(BaseModel):
    id: int
    name: str
    description: str | None = None
    script: str
    param_schema: dict | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
